joint_state_publisher/joint_state_publisher/joint_state_publisher.py

```diff
@@ -416,11 +416,11 @@ def main():
 
     try:
         rclpy.spin(jsp)
+        jsp.destroy_node()
```

Comment:
This means `jsp.destroy_node()` won't get called when SIGINT is used, but honestly that's fine. The garbage collector will clean it up eventually, and in normal use this is the "main" function of a  program so the python interpreter is going to terminate anyways.

A slight improvement might be to keep these where they were, but use `rclpy.try_shutdown()` instead of `rclpy.shutdown()`.

Reply:
@sloretz Using `try_shutdown()` is an even more graceful way to do this. I changed it in 9208b8b